### PR TITLE
[DOCS] Add data streams to security docs

### PR DESCRIPTION
**Pull request by jrodewig**
Makes the existing security docs aware of data streams.
Relates to #58381 and #59139.

### CODE REVIEW

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by jrodewig** [comment]
@elasticmachine  update branch

**Comment by jrodewig** [comment]
@elasticmachine update branch

File: x-pack/docs/en/security/authorization/document-level-security.asciidoc [hunk 1]
```diff
@@ -7,12 +7,12 @@ In particular, it restricts which documents can be accessed from document-based
 read APIs. 
 
 To enable document level security, you use a query to specify the documents that 
-each role can access. The document query is associated with a particular index 
+each role can access. The document query is associated with a particular data stream, index, 
 or index pattern and operates in conjunction with the privileges specified for 
-the indices.
+the data streams and indices.
```
Comment (danhermann) [inline x-pack/docs/en/security/authorization/document-level-security.asciidoc:12]: Minor: I suppose it's now "data stream, index, and index _or data stream_ pattern."

**Review by danhermann — APPROVED** [review]
Left a minor comment, but LGTM otherwise.

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Pull request merged**